tests/unit_tests/Video_test.cc

```diff
@@ -659,4 +659,33 @@ TEST_F(VideoTest, CropWrite)
         print_exception(e);
         ASSERT_TRUE(false);
     }
+}
+
+TEST_F(VideoTest, KeyFrameExtractionSuccess)
+{
+    try {
+        VCL::VideoTest video_data(_video_path_mp4_h264);
+
+        auto key_frame_list = video_data.get_key_frame_list();
+
+        ASSERT_FALSE(key_frame_list.empty());
```

Comment:
same here, it would be nice key_frame_list has the right number of keyframes for that video